libusb/descriptor.c

```diff
@@ -274,27 +275,47 @@ static int parse_interface(libusb_context *ctx,
 
 		if (ifp->bNumEndpoints > 0) {
 			struct libusb_endpoint_descriptor *endpoint;
-			uint8_t i;
+
+			/* Endpoint desciptors reported */
+			if_desc_eps = ifp->bNumEndpoints;
 
 			endpoint = calloc(ifp->bNumEndpoints, sizeof(*endpoint));
```

Comment:
Aren't we gonna overflow this allocation if we end up with more endpoints than ifp->bNumEndpoints?

Reply:
Ah good catch. For my product it shouldn't be a problem, but for a product with + 2 more endpoints it could. I'll look over the whole change and update

Reply:
You probably have your own ideas about how to deal with it, anyway I would have looked at making parse_endpoint() conditionally a read-only function (e.g. when passed a NULL buffer) and call it once to count endpoints, allocate, and call it again.

Reply:
Yes, I am working on a different solution. I've found that my suggestion has more problems than the one you pointed out. The new idea is to keep the for loop from before and only parse the amount of endpoint provided by bNumEndpoints and then after, check if we're still pointing at an endpoint descriptor we're not expecting. If so we skip it and point to the next "proper" descriptor. 
I need to do some more testing before I push this suggestion